backups/snapshots/2025-08-30_15-46-05/proxy_manager.py

```
@@ -0,0 +1,42 @@
+        for proxy, block_until in list(self.blocked_proxies.items()):
+            if now > block_until:
+                self.available_proxies.add(proxy)
+                del self.blocked_proxies[proxy]
+
+    def get_proxy(self):
+        """Returns an available proxy, or None if no proxies are available."""
+        self._clean_blocked_proxies()
+        if not self.available_proxies:
+            return None
+        
+        proxy = random.choice(list(self.available_proxies))
```

Ruff (0.12.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)